src/main/java/com/example/bookshop/controller/BookController.java

```diff
@@ -49,4 +51,9 @@ public BookDto updateBookById(@PathVariable Long id,
     public void delete(@PathVariable Long id) {
         bookService.deleteById(id);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> searchBooks(@RequestParam Map<String, String> params) {
```

Comment:
IMO it's better to use a class with statically defined parameters instead of dynamic map. Map has some advantages too, but by using class you are more specific about allowed values. Also, later, when adding Swagger support, the allowed fields will be replicated in the documentation.